### ISSUE TITLE
Usage of abstract can make the compiler generate no C code without error

### ISSUE THREAD

**Issue by antoyo** [githwxi/ATS-Postiats#241]
Hi.
The following code:
```ats
#define ATS_DYNLOADFLAG 0

absvt@ype A

vtypedef D = [l1:addr] (!A@l1 | ptr l1) -> void

vtypedef B = D

vtypedef C = B

extern fun func(arg: !C): int

assume A = B

implement func(arg) =
    0

implement main0() = ()
```
causes the compiler to generate an empty C file without printing any error message.
Can you please fix this?
Do you know why this code is wrong?
Thanks.

**Comment by githwxi** [comment]
This code is not supported because 'A' is recursively defined, that is, the definition of 'A' makes use of 'A'.

**Comment by antoyo** [comment]
Yeah, but `A` makes use a `A` behind a pointer, behind a function pointer, so I guess that could be supported.

Any in case, that should produce a meaningful error, because not producing code is misleading.

**Comment by githwxi** [comment]
Checking for recursive definition is not currently performed by patsopt. The crash you experienced was due to the expansion of A, which is not terminating.

Instead of using 'assume', you may want to use casting to circumvent the issue.
